Print YANG tree

mzagozen opened the pull request:
I initially added the `DNode.prtree()`, but like @plajjan noted in #84 the transaction to DNode discards module information. I then added `SchemaNode.prtree()` and used it on an intermediate result from `yang.compile()`.

Closes #84

plajjan commented on src/yang.act:
```diff
@@ -55,8 +55,13 @@ extension NameRevMap[T](Collection[(str, (?str, T))]):
 
-def compile(yang_sources: list[str]) -> yang.schema.DRoot:
-    """Parse and compile a set of YANG modules and submodules into a schema tree
+def compile_to_dmodule(yang_sources: list[str]) -> list[yang.schema.DModule]:
```
I think we should strive and encourage modularity and composability by pipelining. SchemaNode is the natural format for transforms etc. Conversion to DNode should only happen at the very end, so I suggest that we rename this function and let it return SchemaNode, or I guess yang.schema.Module.
```suggestion
def compile_modules(yang_sources: list[str]) -> list[yang.schema.Module]:
```

then we conversion to DNode and merge into a single root is in another function, possibly just inlined in the `compile` function

plajjan commented:
At times I ask myself if it was really a clever idea to create DNode. The purpose is of course to have a simple tree for working with YANG-modeled **data** and what better way than to ensure the simple tree through type correctness since there simply are no subclasses for augments or other abstract schema concepts. But then again, we end up with two types of trees, SchemaNode and DNode, that largely overlap and methods need to be duplicated between them unless it's acceptable to only have functionality in one. If we are careful, we can naturally achieve the same thing with just SchemaNode, i.e. we can still compile to a derived schema which only contains data nodes, just like DNode, but we can use a subset of the SchemaNode subclassses for it instead of DNode.

The benefits are clear. prsrc, prtree and similar on SchemaNode always work and are readily available regardless of what stage in the compilation process we are at.

Oh well, perhaps we'll have to struggle some more. I do suppose that a little duplicated work here might still benefit all other users of this library.

anyhoo, I think prtree() should exist on SchemaNode primarily, possibly not at all on DNode, but if we want, we can duplicate it there. Having it on SchemaNode opens up to printing trees under processing, between transforms and perhaps we can also condition expansion, such that we can expand augments or groupings but not types, which would make the output look better.

Also, are these really short names a good idea? I know I started it, but print_tree() does look more intuitive.

mzagozen commented:
> anyhoo, I think prtree() should exist on SchemaNode primarily, possibly not at all on DNode, but if we want, we can duplicate it there. Having it on SchemaNode opens up to printing trees under processing, between transforms and perhaps we can also condition expansion, such that we can expand augments or groupings but not types, which would make the output look better.

I agree, primarily print_tree() should work on SchemaNode. I added the option to *compile* SchemaNode without resolving types, just expand children, which I think is the default in pyang too. I already wrote the code for DNode.print_tree() before, so I will just leave it in.

> Also, are these really short names a good idea? I know I started it, but print_tree() does look more intuitive.

Yes, renamed.
